remote/remote_test.go

```diff
@@ -1595,5 +1658,47 @@ func testImage(t *testing.T, when spec.G, it spec.S) {
 				h.AssertEq(t, image.CheckReadWriteAccess(), false)
 			})
 		})
+
+		when("using custom handler", func() {
+			it.Before(func() {
+				os.Setenv("DOCKER_CONFIG", customRegistry.DockerDirectory)
+			})
+
+			it.After(func() {
+				os.Setenv("DOCKER_CONFIG", dockerRegistry.DockerDirectory)
+			})
+
+			when("image has read/write access", func() {
+				it("returns true", func() {
+					image, err := remote.NewImage(customRegistry.RepoName(readWriteImage), authn.DefaultKeychain)
+					h.AssertNil(t, err)
+					h.AssertEq(t, image.CheckReadWriteAccess(), true)
+				})
+			})
+
+			when("image has read access but no write access", func() {
+				it("returns false", func() {
+					image, err := remote.NewImage(customRegistry.RepoName(readNoWriteImage), authn.DefaultKeychain)
+					h.AssertNil(t, err)
+					h.AssertEq(t, image.CheckReadWriteAccess(), false)
+				})
+			})
+
+			when("image doesn't have read access but has write access", func() {
+				it("returns true", func() {
+					image, err := remote.NewImage(customRegistry.RepoName(noReadWriteImage), authn.DefaultKeychain)
+					h.AssertNil(t, err)
+					h.AssertEq(t, image.CheckReadWriteAccess(), false)
+				})
+			})
+
+			when("image doesn't have read nor write access", func() {
+				it("returns false", func() {
+					image, err := remote.NewImage(customRegistry.RepoName(noReadNorWriteImage), authn.DefaultKeychain)
+					h.AssertNil(t, err)
+					h.AssertEq(t, image.CheckReadWriteAccess(), false)
```

Comment:
Should we check read access here?

Reply:
Thinking a lit bit more about it, with these new docker registry handlers, I can actually cut in half the code and added the assertions for `CheckReadWriteAccess()` and `CheckReadAccess()` in the same test scenarios. Would it help on making the test cases more readable?

Reply:
Oh, maybe! Up to you...